


<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html id="htmlId">
<head>
  <title>Coverage Report :: DefaultErrorHandler</title>
  <style type="text/css">
    @import "../../.css/coverage.css";
  </style>
</head>

<body>
<div class="header"></div>

<div class="content">
<div class="breadCrumbs">
    [ <a href="../../index.html">all classes</a> ]
    [ <a href="../index.html">com.sun.org.apache.xml.internal.utils</a> ]
</div>

<h1>Coverage Summary for Class: DefaultErrorHandler (com.sun.org.apache.xml.internal.utils)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">DefaultErrorHandler</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 14)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/ 65)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<div class="sourceCode"><i>1</i>&nbsp;/*
<i>2</i>&nbsp; * Copyright (c) 2007, 2016, Oracle and/or its affiliates. All rights reserved.
<i>3</i>&nbsp; * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
<i>4</i>&nbsp; */
<i>5</i>&nbsp;/*
<i>6</i>&nbsp; * Copyright 1999-2004 The Apache Software Foundation.
<i>7</i>&nbsp; *
<i>8</i>&nbsp; * Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);
<i>9</i>&nbsp; * you may not use this file except in compliance with the License.
<i>10</i>&nbsp; * You may obtain a copy of the License at
<i>11</i>&nbsp; *
<i>12</i>&nbsp; *     http://www.apache.org/licenses/LICENSE-2.0
<i>13</i>&nbsp; *
<i>14</i>&nbsp; * Unless required by applicable law or agreed to in writing, software
<i>15</i>&nbsp; * distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
<i>16</i>&nbsp; * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
<i>17</i>&nbsp; * See the License for the specific language governing permissions and
<i>18</i>&nbsp; * limitations under the License.
<i>19</i>&nbsp; */
<i>20</i>&nbsp;/*
<i>21</i>&nbsp; * $Id: DefaultErrorHandler.java,v 1.2.4.1 2005/09/15 08:15:43 suresh_emailid Exp $
<i>22</i>&nbsp; */
<i>23</i>&nbsp;package com.sun.org.apache.xml.internal.utils;
<i>24</i>&nbsp;
<i>25</i>&nbsp;import java.io.PrintStream;
<i>26</i>&nbsp;import java.io.PrintWriter;
<i>27</i>&nbsp;
<i>28</i>&nbsp;import javax.xml.transform.ErrorListener;
<i>29</i>&nbsp;import javax.xml.transform.SourceLocator;
<i>30</i>&nbsp;import javax.xml.transform.TransformerException;
<i>31</i>&nbsp;
<i>32</i>&nbsp;import com.sun.org.apache.xml.internal.res.XMLErrorResources;
<i>33</i>&nbsp;import com.sun.org.apache.xml.internal.res.XMLMessages;
<i>34</i>&nbsp;
<i>35</i>&nbsp;import org.xml.sax.ErrorHandler;
<i>36</i>&nbsp;import org.xml.sax.SAXException;
<i>37</i>&nbsp;import org.xml.sax.SAXParseException;
<i>38</i>&nbsp;
<i>39</i>&nbsp;
<i>40</i>&nbsp;/**
<i>41</i>&nbsp; * Implement SAX error handler for default reporting.
<i>42</i>&nbsp; * @xsl.usage general
<i>43</i>&nbsp; */
<i>44</i>&nbsp;public class DefaultErrorHandler implements ErrorHandler, ErrorListener
<i>45</i>&nbsp;{
<i>46</i>&nbsp;  PrintWriter m_pw;
<i>47</i>&nbsp;
<i>48</i>&nbsp;  /**
<i>49</i>&nbsp;   * if this flag is set to true, we will rethrow the exception on
<i>50</i>&nbsp;   * the error() and fatalError() methods. If it is false, the errors
<i>51</i>&nbsp;   * are reported to System.err.
<i>52</i>&nbsp;   */
<b class="nc"><i>53</i>&nbsp;  boolean m_throwExceptionOnError = true;</b>
<i>54</i>&nbsp;
<i>55</i>&nbsp;  /**
<i>56</i>&nbsp;   * Constructor DefaultErrorHandler
<i>57</i>&nbsp;   */
<i>58</i>&nbsp;  public DefaultErrorHandler(PrintWriter pw)
<b class="nc"><i>59</i>&nbsp;  {</b>
<b class="nc"><i>60</i>&nbsp;    m_pw = pw;</b>
<i>61</i>&nbsp;  }
<i>62</i>&nbsp;
<i>63</i>&nbsp;  /**
<i>64</i>&nbsp;   * Constructor DefaultErrorHandler
<i>65</i>&nbsp;   */
<i>66</i>&nbsp;  public DefaultErrorHandler(PrintStream pw)
<b class="nc"><i>67</i>&nbsp;  {</b>
<b class="nc"><i>68</i>&nbsp;    m_pw = new PrintWriter(pw, true);</b>
<i>69</i>&nbsp;  }
<i>70</i>&nbsp;
<i>71</i>&nbsp;  /**
<i>72</i>&nbsp;   * Constructor DefaultErrorHandler
<i>73</i>&nbsp;   */
<i>74</i>&nbsp;  public DefaultErrorHandler()
<i>75</i>&nbsp;  {
<b class="nc"><i>76</i>&nbsp;    this(true);</b>
<i>77</i>&nbsp;  }
<i>78</i>&nbsp;
<i>79</i>&nbsp;  /**
<i>80</i>&nbsp;   * Constructor DefaultErrorHandler
<i>81</i>&nbsp;   */
<i>82</i>&nbsp;  public DefaultErrorHandler(boolean throwExceptionOnError)
<b class="nc"><i>83</i>&nbsp;  {</b>
<b class="nc"><i>84</i>&nbsp;    m_pw = new PrintWriter(System.err, true);</b>
<b class="nc"><i>85</i>&nbsp;    m_throwExceptionOnError = throwExceptionOnError;</b>
<i>86</i>&nbsp;  }
<i>87</i>&nbsp;
<i>88</i>&nbsp;
<i>89</i>&nbsp;  /**
<i>90</i>&nbsp;   * Receive notification of a warning.
<i>91</i>&nbsp;   *
<i>92</i>&nbsp;   * &lt;p&gt;SAX parsers will use this method to report conditions that
<i>93</i>&nbsp;   * are not errors or fatal errors as defined by the XML 1.0
<i>94</i>&nbsp;   * recommendation.  The default behaviour is to take no action.&lt;/p&gt;
<i>95</i>&nbsp;   *
<i>96</i>&nbsp;   * &lt;p&gt;The SAX parser must continue to provide normal parsing events
<i>97</i>&nbsp;   * after invoking this method: it should still be possible for the
<i>98</i>&nbsp;   * application to process the document through to the end.&lt;/p&gt;
<i>99</i>&nbsp;   *
<i>100</i>&nbsp;   * @param exception The warning information encapsulated in a
<i>101</i>&nbsp;   *                  SAX parse exception.
<i>102</i>&nbsp;   * @throws SAXException Any SAX exception, possibly
<i>103</i>&nbsp;   *            wrapping another exception.
<i>104</i>&nbsp;   */
<i>105</i>&nbsp;  public void warning(SAXParseException exception) throws SAXException
<i>106</i>&nbsp;  {
<b class="nc"><i>107</i>&nbsp;    printLocation(m_pw, exception);</b>
<b class="nc"><i>108</i>&nbsp;    m_pw.println(&quot;Parser warning: &quot; + exception.getMessage());</b>
<i>109</i>&nbsp;  }
<i>110</i>&nbsp;
<i>111</i>&nbsp;  /**
<i>112</i>&nbsp;   * Receive notification of a recoverable error.
<i>113</i>&nbsp;   *
<i>114</i>&nbsp;   * &lt;p&gt;This corresponds to the definition of &quot;error&quot; in section 1.2
<i>115</i>&nbsp;   * of the W3C XML 1.0 Recommendation.  For example, a validating
<i>116</i>&nbsp;   * parser would use this callback to report the violation of a
<i>117</i>&nbsp;   * validity constraint.  The default behaviour is to take no
<i>118</i>&nbsp;   * action.&lt;/p&gt;
<i>119</i>&nbsp;   *
<i>120</i>&nbsp;   * &lt;p&gt;The SAX parser must continue to provide normal parsing events
<i>121</i>&nbsp;   * after invoking this method: it should still be possible for the
<i>122</i>&nbsp;   * application to process the document through to the end.  If the
<i>123</i>&nbsp;   * application cannot do so, then the parser should report a fatal
<i>124</i>&nbsp;   * error even if the XML 1.0 recommendation does not require it to
<i>125</i>&nbsp;   * do so.&lt;/p&gt;
<i>126</i>&nbsp;   *
<i>127</i>&nbsp;   * @param exception The error information encapsulated in a
<i>128</i>&nbsp;   *                  SAX parse exception.
<i>129</i>&nbsp;   * @throws SAXException Any SAX exception, possibly
<i>130</i>&nbsp;   *            wrapping another exception.
<i>131</i>&nbsp;   */
<i>132</i>&nbsp;  public void error(SAXParseException exception) throws SAXException
<i>133</i>&nbsp;  {
<i>134</i>&nbsp;    //printLocation(exception);
<i>135</i>&nbsp;    // m_pw.println(exception.getMessage());
<i>136</i>&nbsp;
<b class="nc"><i>137</i>&nbsp;    throw exception;</b>
<i>138</i>&nbsp;  }
<i>139</i>&nbsp;
<i>140</i>&nbsp;  /**
<i>141</i>&nbsp;   * Receive notification of a non-recoverable error.
<i>142</i>&nbsp;   *
<i>143</i>&nbsp;   * &lt;p&gt;This corresponds to the definition of &quot;fatal error&quot; in
<i>144</i>&nbsp;   * section 1.2 of the W3C XML 1.0 Recommendation.  For example, a
<i>145</i>&nbsp;   * parser would use this callback to report the violation of a
<i>146</i>&nbsp;   * well-formedness constraint.&lt;/p&gt;
<i>147</i>&nbsp;   *
<i>148</i>&nbsp;   * &lt;p&gt;The application must assume that the document is unusable
<i>149</i>&nbsp;   * after the parser has invoked this method, and should continue
<i>150</i>&nbsp;   * (if at all) only for the sake of collecting addition error
<i>151</i>&nbsp;   * messages: in fact, SAX parsers are free to stop reporting any
<i>152</i>&nbsp;   * other events once this method has been invoked.&lt;/p&gt;
<i>153</i>&nbsp;   *
<i>154</i>&nbsp;   * @param exception The error information encapsulated in a
<i>155</i>&nbsp;   *                  SAX parse exception.
<i>156</i>&nbsp;   * @throws SAXException Any SAX exception, possibly
<i>157</i>&nbsp;   *            wrapping another exception.
<i>158</i>&nbsp;   */
<i>159</i>&nbsp;  public void fatalError(SAXParseException exception) throws SAXException
<i>160</i>&nbsp;  {
<i>161</i>&nbsp;    // printLocation(exception);
<i>162</i>&nbsp;    // m_pw.println(exception.getMessage());
<i>163</i>&nbsp;
<b class="nc"><i>164</i>&nbsp;    throw exception;</b>
<i>165</i>&nbsp;  }
<i>166</i>&nbsp;
<i>167</i>&nbsp;  /**
<i>168</i>&nbsp;   * Receive notification of a warning.
<i>169</i>&nbsp;   *
<i>170</i>&nbsp;   * &lt;p&gt;SAX parsers will use this method to report conditions that
<i>171</i>&nbsp;   * are not errors or fatal errors as defined by the XML 1.0
<i>172</i>&nbsp;   * recommendation.  The default behaviour is to take no action.&lt;/p&gt;
<i>173</i>&nbsp;   *
<i>174</i>&nbsp;   * &lt;p&gt;The SAX parser must continue to provide normal parsing events
<i>175</i>&nbsp;   * after invoking this method: it should still be possible for the
<i>176</i>&nbsp;   * application to process the document through to the end.&lt;/p&gt;
<i>177</i>&nbsp;   *
<i>178</i>&nbsp;   * @param exception The warning information encapsulated in a
<i>179</i>&nbsp;   *                  SAX parse exception.
<i>180</i>&nbsp;   * @throws javax.xml.transform.TransformerException Any SAX exception, possibly
<i>181</i>&nbsp;   *            wrapping another exception.
<i>182</i>&nbsp;   * @see javax.xml.transform.TransformerException
<i>183</i>&nbsp;   */
<i>184</i>&nbsp;  public void warning(TransformerException exception) throws TransformerException
<i>185</i>&nbsp;  {
<b class="nc"><i>186</i>&nbsp;    printLocation(m_pw, exception);</b>
<i>187</i>&nbsp;
<b class="nc"><i>188</i>&nbsp;    m_pw.println(exception.getMessage());</b>
<i>189</i>&nbsp;  }
<i>190</i>&nbsp;
<i>191</i>&nbsp;  /**
<i>192</i>&nbsp;   * Receive notification of a recoverable error.
<i>193</i>&nbsp;   *
<i>194</i>&nbsp;   * &lt;p&gt;This corresponds to the definition of &quot;error&quot; in section 1.2
<i>195</i>&nbsp;   * of the W3C XML 1.0 Recommendation.  For example, a validating
<i>196</i>&nbsp;   * parser would use this callback to report the violation of a
<i>197</i>&nbsp;   * validity constraint.  The default behaviour is to take no
<i>198</i>&nbsp;   * action.&lt;/p&gt;
<i>199</i>&nbsp;   *
<i>200</i>&nbsp;   * &lt;p&gt;The SAX parser must continue to provide normal parsing events
<i>201</i>&nbsp;   * after invoking this method: it should still be possible for the
<i>202</i>&nbsp;   * application to process the document through to the end.  If the
<i>203</i>&nbsp;   * application cannot do so, then the parser should report a fatal
<i>204</i>&nbsp;   * error even if the XML 1.0 recommendation does not require it to
<i>205</i>&nbsp;   * do so.&lt;/p&gt;
<i>206</i>&nbsp;   *
<i>207</i>&nbsp;   * @param exception The error information encapsulated in a
<i>208</i>&nbsp;   *                  SAX parse exception.
<i>209</i>&nbsp;   * @throws javax.xml.transform.TransformerException Any SAX exception, possibly
<i>210</i>&nbsp;   *            wrapping another exception.
<i>211</i>&nbsp;   * @see javax.xml.transform.TransformerException
<i>212</i>&nbsp;   */
<i>213</i>&nbsp;  public void error(TransformerException exception) throws TransformerException
<i>214</i>&nbsp;  {
<i>215</i>&nbsp;    // If the m_throwExceptionOnError flag is true, rethrow the exception.
<i>216</i>&nbsp;    // Otherwise report the error to System.err.
<b class="nc"><i>217</i>&nbsp;    if (m_throwExceptionOnError)</b>
<b class="nc"><i>218</i>&nbsp;      throw exception;</b>
<i>219</i>&nbsp;    else
<i>220</i>&nbsp;    {
<b class="nc"><i>221</i>&nbsp;      printLocation(m_pw, exception);</b>
<b class="nc"><i>222</i>&nbsp;      m_pw.println(exception.getMessage());</b>
<i>223</i>&nbsp;    }
<i>224</i>&nbsp;  }
<i>225</i>&nbsp;
<i>226</i>&nbsp;  /**
<i>227</i>&nbsp;   * Receive notification of a non-recoverable error.
<i>228</i>&nbsp;   *
<i>229</i>&nbsp;   * &lt;p&gt;This corresponds to the definition of &quot;fatal error&quot; in
<i>230</i>&nbsp;   * section 1.2 of the W3C XML 1.0 Recommendation.  For example, a
<i>231</i>&nbsp;   * parser would use this callback to report the violation of a
<i>232</i>&nbsp;   * well-formedness constraint.&lt;/p&gt;
<i>233</i>&nbsp;   *
<i>234</i>&nbsp;   * &lt;p&gt;The application must assume that the document is unusable
<i>235</i>&nbsp;   * after the parser has invoked this method, and should continue
<i>236</i>&nbsp;   * (if at all) only for the sake of collecting addition error
<i>237</i>&nbsp;   * messages: in fact, SAX parsers are free to stop reporting any
<i>238</i>&nbsp;   * other events once this method has been invoked.&lt;/p&gt;
<i>239</i>&nbsp;   *
<i>240</i>&nbsp;   * @param exception The error information encapsulated in a
<i>241</i>&nbsp;   *                  SAX parse exception.
<i>242</i>&nbsp;   * @throws javax.xml.transform.TransformerException Any SAX exception, possibly
<i>243</i>&nbsp;   *            wrapping another exception.
<i>244</i>&nbsp;   * @see javax.xml.transform.TransformerException
<i>245</i>&nbsp;   */
<i>246</i>&nbsp;  public void fatalError(TransformerException exception) throws TransformerException
<i>247</i>&nbsp;  {
<i>248</i>&nbsp;    // If the m_throwExceptionOnError flag is true, rethrow the exception.
<i>249</i>&nbsp;    // Otherwise report the error to System.err.
<b class="nc"><i>250</i>&nbsp;    if (m_throwExceptionOnError)</b>
<b class="nc"><i>251</i>&nbsp;      throw exception;</b>
<i>252</i>&nbsp;    else
<i>253</i>&nbsp;    {
<b class="nc"><i>254</i>&nbsp;      printLocation(m_pw, exception);</b>
<b class="nc"><i>255</i>&nbsp;      m_pw.println(exception.getMessage());</b>
<i>256</i>&nbsp;    }
<i>257</i>&nbsp;  }
<i>258</i>&nbsp;
<i>259</i>&nbsp;  public static void ensureLocationSet(TransformerException exception)
<i>260</i>&nbsp;  {
<i>261</i>&nbsp;    // SourceLocator locator = exception.getLocator();
<b class="nc"><i>262</i>&nbsp;    SourceLocator locator = null;</b>
<b class="nc"><i>263</i>&nbsp;    Throwable cause = exception;</b>
<i>264</i>&nbsp;
<i>265</i>&nbsp;    // Try to find the locator closest to the cause.
<i>266</i>&nbsp;    do
<i>267</i>&nbsp;    {
<b class="nc"><i>268</i>&nbsp;      if(cause instanceof SAXParseException)</b>
<i>269</i>&nbsp;      {
<b class="nc"><i>270</i>&nbsp;        locator = new SAXSourceLocator((SAXParseException)cause);</b>
<i>271</i>&nbsp;      }
<b class="nc"><i>272</i>&nbsp;      else if (cause instanceof TransformerException)</b>
<i>273</i>&nbsp;      {
<b class="nc"><i>274</i>&nbsp;        SourceLocator causeLocator = ((TransformerException)cause).getLocator();</b>
<b class="nc"><i>275</i>&nbsp;        if(null != causeLocator)</b>
<b class="nc"><i>276</i>&nbsp;          locator = causeLocator;</b>
<i>277</i>&nbsp;      }
<i>278</i>&nbsp;
<b class="nc"><i>279</i>&nbsp;      if(cause instanceof TransformerException)</b>
<b class="nc"><i>280</i>&nbsp;        cause = ((TransformerException)cause).getCause();</b>
<b class="nc"><i>281</i>&nbsp;      else if(cause instanceof SAXException)</b>
<b class="nc"><i>282</i>&nbsp;        cause = ((SAXException)cause).getException();</b>
<i>283</i>&nbsp;      else
<b class="nc"><i>284</i>&nbsp;        cause = null;</b>
<i>285</i>&nbsp;    }
<b class="nc"><i>286</i>&nbsp;    while(null != cause);</b>
<i>287</i>&nbsp;
<b class="nc"><i>288</i>&nbsp;    exception.setLocator(locator);</b>
<i>289</i>&nbsp;  }
<i>290</i>&nbsp;
<i>291</i>&nbsp;  public static void printLocation(PrintStream pw, TransformerException exception)
<i>292</i>&nbsp;  {
<b class="nc"><i>293</i>&nbsp;    printLocation(new PrintWriter(pw), exception);</b>
<i>294</i>&nbsp;  }
<i>295</i>&nbsp;
<i>296</i>&nbsp;  public static void printLocation(java.io.PrintStream pw, org.xml.sax.SAXParseException exception)
<i>297</i>&nbsp;  {
<b class="nc"><i>298</i>&nbsp;    printLocation(new PrintWriter(pw), exception);</b>
<i>299</i>&nbsp;  }
<i>300</i>&nbsp;
<i>301</i>&nbsp;  public static void printLocation(PrintWriter pw, Throwable exception)
<i>302</i>&nbsp;  {
<b class="nc"><i>303</i>&nbsp;    SourceLocator locator = null;</b>
<b class="nc"><i>304</i>&nbsp;    Throwable cause = exception;</b>
<i>305</i>&nbsp;
<i>306</i>&nbsp;    // Try to find the locator closest to the cause.
<i>307</i>&nbsp;    do
<i>308</i>&nbsp;    {
<b class="nc"><i>309</i>&nbsp;      if(cause instanceof SAXParseException)</b>
<i>310</i>&nbsp;      {
<b class="nc"><i>311</i>&nbsp;        locator = new SAXSourceLocator((SAXParseException)cause);</b>
<i>312</i>&nbsp;      }
<b class="nc"><i>313</i>&nbsp;      else if (cause instanceof TransformerException)</b>
<i>314</i>&nbsp;      {
<b class="nc"><i>315</i>&nbsp;        SourceLocator causeLocator = ((TransformerException)cause).getLocator();</b>
<b class="nc"><i>316</i>&nbsp;        if(null != causeLocator)</b>
<b class="nc"><i>317</i>&nbsp;          locator = causeLocator;</b>
<i>318</i>&nbsp;      }
<b class="nc"><i>319</i>&nbsp;      if(cause instanceof TransformerException)</b>
<b class="nc"><i>320</i>&nbsp;        cause = ((TransformerException)cause).getCause();</b>
<b class="nc"><i>321</i>&nbsp;      else if(cause instanceof WrappedRuntimeException)</b>
<b class="nc"><i>322</i>&nbsp;        cause = ((WrappedRuntimeException)cause).getException();</b>
<b class="nc"><i>323</i>&nbsp;      else if(cause instanceof SAXException)</b>
<b class="nc"><i>324</i>&nbsp;        cause = ((SAXException)cause).getException();</b>
<i>325</i>&nbsp;      else
<b class="nc"><i>326</i>&nbsp;        cause = null;</b>
<i>327</i>&nbsp;    }
<b class="nc"><i>328</i>&nbsp;    while(null != cause);</b>
<i>329</i>&nbsp;
<b class="nc"><i>330</i>&nbsp;    if(null != locator)</b>
<i>331</i>&nbsp;    {
<i>332</i>&nbsp;      // m_pw.println(&quot;Parser fatal error: &quot;+exception.getMessage());
<b class="nc"><i>333</i>&nbsp;      String id = (null != locator.getPublicId() )</b>
<b class="nc"><i>334</i>&nbsp;                  ? locator.getPublicId()</b>
<b class="nc"><i>335</i>&nbsp;                    : (null != locator.getSystemId())</b>
<b class="nc"><i>336</i>&nbsp;                      ? locator.getSystemId() : XMLMessages.createXMLMessage(XMLErrorResources.ER_SYSTEMID_UNKNOWN, null); //&quot;SystemId Unknown&quot;;</b>
<i>337</i>&nbsp;
<b class="nc"><i>338</i>&nbsp;      pw.print(id + &quot;; &quot; +XMLMessages.createXMLMessage(&quot;line&quot;, null) + locator.getLineNumber()</b>
<b class="nc"><i>339</i>&nbsp;                         + &quot;; &quot; +XMLMessages.createXMLMessage(&quot;column&quot;, null) + locator.getColumnNumber()+&quot;; &quot;);</b>
<b class="nc"><i>340</i>&nbsp;    }</b>
<i>341</i>&nbsp;    else
<b class="nc"><i>342</i>&nbsp;      pw.print(&quot;(&quot;+XMLMessages.createXMLMessage(XMLErrorResources.ER_LOCATION_UNKNOWN, null)+&quot;)&quot;);</b>
<i>343</i>&nbsp;  }
<i>344</i>&nbsp;}
</div>
</div>

<div class="footer">
    
    <div style="float:right;">generated on 2018-05-11 02:56</div>
</div>
</body>
</html>
